### PR TITLE
Binding Border.StrokeShape stopped working - fix

### LINKED ISSUE #22549: Binding Border.StrokeShape stopped working in 8.0.60-ci.net8.10665
### Description

Hi,
after upgrade the nuget nightlys my binding stopped working. Sample code:
```XAML
      <Border x:Name="myBorder"
              Margin="0,10,0,0"
              BackgroundColor="Blue"
              StrokeShape="{Binding Source={RelativeSource AncestorType={x:Type viewmodels:TestViewModel}}, Path=RoundedRect}"
              Padding="0"
              StrokeThickness="0">
        <!--<Border.StrokeShape>
          <RoundRectangle CornerRadius="{Binding Source={RelativeSource AncestorType={x:Type viewmodels:TestViewModel}}, Path=Radius}" />
        </Border.StrokeShape>-->
```
I init the StrokeShape wiith a CornerRadius of 4 and on the button click i create a new CornerRadius of 12. The change was not applied. 
```C#
        private void OnCounterClicked(object sender, EventArgs e)
        {
            count++;

            if (count == 1)
                CounterBtn.Text = $"Clicked {count} time";
            else
                CounterBtn.Text = $"Clicked {count} times";
            var model = (TestViewModel)BindingContext;
            var r = model.RoundedRect;
            r.CornerRadius = new CornerRadius(12, 12, 12, 12);
            model.RoundedRect = r;
            //SemanticScreenReader.Announce(CounterBtn.Text);
        }
```

### Steps to Reproduce

Start demo app und click button. Corner should be 12, but it's still 4.
Open csproj file and change nightl version to: 8.0.60-ci.net8.10656. Clean, build and start app again. Click the button, now the corner was changed.
Latest working version: 8.0.60-ci.net8.10656
First broken version: 8.0.60-ci.net8.10665
[MauiApp8.zip](https://github.com/dotnet/maui/files/15390650/MauiApp8.zip)

The StrokeShape was updated if Hot Reload is triggered or the device was rotated.

### Link to public reproduction project repository

_No response_

### Version with bug

Nightly / CI build (Please specify exact version)

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
Fixes https://github.com/dotnet/maui/issues/22549

<video src="https://github.com/dotnet/maui/assets/42434498/e1c104bb-6c20-4b51-9a04-0ccc100e5616"/>

### CODE REVIEW

**Comment by mattleibow** [comment]
@kubaflo is this just a revert?

**Comment by kubaflo** [comment]
@mattleibow it is not a revert but a small correction to this: https://github.com/dotnet/maui/pull/21197#discussion_r1580983816

**Comment by mattleibow** [comment]
Ah, nice.

I wonder if it is worth adding a test that when you click a button the border changes to a width and then back again to make sure that updates/bindings can set it back to 0.

For example, assume you have a bindable property that changes at runtime. Theoretically going from 0, screenshot, 1, 0, screenshot should have the same result.

Similarly, maybe add a new test for the corner radius. This should have been caught by CI, so we got some holes in our testing.

**Comment by kubaflo** [comment]
@mattleibow I've added tests :)

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22549.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		{
+			App.WaitForElement("button");
+
+			//Border should have radius
+			VerifyScreenshot();
+
+			App.Click("button");
+
+			//The test passes if border radius is equal to 0
+			VerifyScreenshot();
```
Comment (mattleibow) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22549.cs:24]: These may need to have a custom filename since they are different.

**Review by mattleibow — COMMENTED** [review]
The tests are great! Love the Xaml one too.

Just a comment on the screenshot filename.

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22549.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22549.cs]: Right, added different names.

**Review by PureWeen — APPROVED** [review]
